### ISSUE TITLE
Support for More Loss Functions

### ISSUE THREAD

**Issue by MohamedAliRashad** [pytorch/pytorch#41142]
## 🚀 Feature
Supporting more loss functions like `FocalLoss` or `DICE` natively will be a nice addition, especially as their already implementations to them out there.

## Motivation

Native support for more loss functions will reduce boilerplate code.

## Pitch

Review and addition of loss functions found [Here](https://github.com/JunMa11/SegLoss)

## Alternatives

There is implementations on PyTorch Forums for some of the loss functions mentioned above but because they are not reviewed they sometimes have big flaws.

**Comment by MohamedAliRashad** [comment]
https://github.com/pytorch/pytorch/issues/1249 Dice Loss issue here
https://github.com/pytorch/pytorch/issues/35882 Focal Loss issue here

**Comment by alaydshah** [comment]
Can I take up this issue?

**Comment by MohamedAliRashad** [comment]
@alaydshah Go ahead :')
